src/config/mongoose_config_parser_toml.erl

```diff
@@ -1365,21 +1365,44 @@ process_acl([ACLName, _|Path], Content) ->
     [acl:to_record(host(Path), b2a(ACLName), acl_data(Content))].
 
 -spec acl_data(toml_value()) -> option().
-acl_data(Content) when is_map(Content) ->
-    case maps:to_list(Content) of
-        [{Key, Values}] when is_list(Values) ->
-            list_to_tuple([b2a(Key) | Values]);
-        [{Key, Value}] when is_binary(Value) ->
-            {b2a(Key), Value}
-    end;
-acl_data(Value) when is_binary(Value) -> b2a(Value).
+acl_data(<<"all">>) -> all;
+acl_data(<<"none">>) -> none;
+acl_data(M) ->
+    {AclName, AclKeys} = find_acl(M, lists:sort(maps:keys(M)), acl_keys()),
+    list_to_tuple([AclName | lists:map(fun(K) -> maps:get(K, M) end, AclKeys)]).
+
+find_acl(M, SortedMapKeys, [{AclName, AclKeys}|Rest]) ->
+    case lists:sort(AclKeys) of
+        SortedMapKeys -> {AclName, AclKeys};
+        _ -> find_acl(M, SortedMapKeys, Rest)
+    end.
+
+acl_keys() ->
```

Comment:
In `acl` module there is also `resource`, isn't it missing?

Reply:
Good catch!